src/rez/utils/platform_.py

```diff
@@ -17,20 +16,68 @@ class Platform(object):
     """
     name = None
 
-    def __init__(self):
-        pass
+    def __init__(self, platform_map=None):
+        """
+        Construct platform
+
+        Args:
+            platform_map: Force explicit platform map
+        """
+        self._platform_map = platform_map
+
+    def _platform_mapped(self, key, value):
+        """Looks the value up within the _platform_map of the instance or the
+        config.platform_map dictionary.
+
+        Note that there is no guaranteed order within the dictionary evaluation.
+        Only the first matching regular expression is being used.
+        For example:
+
+        config.platform_map = {
+            "os": {
+                r"Scientific Linux-(.*)": r"Scientific-\1",    # Scientific Linux-x.x -> Scientific-x.x
+                r"Ubuntu-14.\d": r"Ubuntu-14",                 # Any Ubuntu-14.x      -> Ubuntu-14
+            },
+            "arch": {
+                "x86_64": "64bit",                             # Maps both x86_64 and amd64 -> 64bit (don't)
+                "amd64": "64bit",
+            },
+        }
+
+        Args:
+            key (str): key to use in the platform_map
+            value (str): the value to map
+
+        Returns:
+            platform mapped value or original value if no map is found
+
+        """
+        platform_map = self._platform_map
+        if platform_map is None:
+            # Since platform is being used within config lazy import config to
+            # prevent circular dependencies
+            from rez.config import config
+            platform_map = config.platform_map
+
+        # The function name is used as primary key
+        entry = platform_map.get(key)
+        if entry:
+            for key, value in entry.iteritems():
```

Comment:
for py3, change iteritems to items (represents solve to one of my test case failures.)

Reply:
Fixed. Thanks!